src/schema/tokens/pack.ts

```diff
@@ -0,0 +1,97 @@
+import { z } from "zod";
+import { BigNumberishSchema, PriceSchema, RawDateSchema } from "../shared";
+import { NFTInputOrUriSchema } from "./common";
+import {
+  ERC1155WrappableSchema,
+  ERC20WrappableSchema,
+  ERC721WrappableSchema,
+} from "./common/wrap";
+
+/**
+ * @internal
+ */
+const ERC20RewardSchema = ERC20WrappableSchema.omit({
+  quantity: true,
+}).extend({
+  quantityPerReward: PriceSchema,
+});
+
+/**
+ * @internal
+ */
+const ERC721RewardSchema = ERC721WrappableSchema;
+
+/**
+ * @internal
+ */
+const ERC1155RewardSchema = ERC1155WrappableSchema.omit({
+  quantity: true,
+}).extend({
+  quantityPerReward: BigNumberishSchema,
+});
+
+/**
+ * @internal
+ */
+const ERC20RewardContentsSchema = ERC20RewardSchema.extend({
+  totalRewards: BigNumberishSchema.default("1"),
+});
+
+/**
+ * @internal
+ */
+const ERC721RewardContentsSchema = ERC721RewardSchema;
+
+/**
+ * @internal
+ */
+const ERC1155RewardContentsSchema = ERC1155RewardSchema.extend({
+  totalRewards: BigNumberishSchema.default("1"),
+});
+
+/**
+ * @internal
+ */
+export const PackRewardsSchema = z.object({
+  erc20Rewards: z.array(ERC20RewardSchema).default([]),
+  erc721Rewards: z.array(ERC721RewardSchema).default([]),
+  erc1155Rewards: z.array(ERC1155RewardSchema).default([]),
+});
+
+/**
+ * @internal
+ */
+export const PackRewardsOutputSchema = z.object({
+  erc20Rewards: z.array(ERC20RewardContentsSchema).default([]),
+  erc721Rewards: z.array(ERC721RewardContentsSchema).default([]),
+  erc1155Rewards: z.array(ERC1155RewardContentsSchema).default([]),
```

Comment:
same for this - `erc20Contents` would make sense in both the pack and MW context right?

Reply:
Schema is a bit different for rewards vs MW because of the amounts stuff